File: Anul II/Medii de proiectare si programare/Laborator/lab02/C#/Ubb.BikeContest.Model/src/Race.cs
namespace Ubb.BikeContest.Model;

public class Race : Identifiable<long>
{
    private string _name;
    private int _engineCapacity;

    public Race(long id, string name, int engineCapacity)
    {
        Id = id;
        _name = name;
        _engineCapacity = engineCapacity;
    }

    public string Name
    {
        get => _name;
        set => _name = value;
    }

    public int EngineCapacity
    {
        get => _engineCapacity;
        set => _engineCapacity = value;
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab11/java/model/src/main/java/com/ubb/model/Race.java
package com.ubb.model;

public class Race extends Identifiable<Long> {
    private String name;
    private Integer engineCapacity;

    {
        setId(0L);
    }
    
    public Race() {

    }

    public Race(String name, Integer engineCapacity) {
        this.name = name;
        this.engineCapacity = engineCapacity;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getEngineCapacity() {
        return engineCapacity;
    }

    public void setEngineCapacity(Integer engineCapacity) {
        this.engineCapacity = engineCapacity;
    }

    @Override
    public String toString() {
        return getName();
    }
}
